sync3/dispatcher.go

```diff
@@ -31,7 +35,8 @@ type Dispatcher struct {
 	jrt              *JoinedRoomsTracker
 	userToReceiver   map[string]Receiver
 	userToReceiverMu *sync.RWMutex
-	latestPos        int64
+	// latestPos is an eventNID, the largest that this dispatcher has seen.
+	latestPos int64
```

Comment:
Aside: a comment on WTF the dispatcher does would be helpful. It sounds like its job is to post-process certain V2Listener messages and re-publish them in a form that is suitable for the caches?

Reply:
Yeah, as per previous convos, the dispatcher can be folded into the Handler at this point.